backend/src/main/scala/com/softwaremill/adopttapir/starter/StarterService.scala

```diff
@@ -1,49 +1,35 @@
 package com.softwaremill.adopttapir.starter
 
-import better.files.File.newTemporaryDirectory
-import better.files.{File => BFile}
 import cats.effect.IO
 import com.softwaremill.adopttapir.logging.FLogging
 import com.softwaremill.adopttapir.metrics.Metrics
 import com.softwaremill.adopttapir.metrics.Metrics.generatedStarterCounter
-import com.softwaremill.adopttapir.template.{GeneratedFile, ProjectGenerator}
-import com.softwaremill.adopttapir.util.ZipArchiver
+import com.softwaremill.adopttapir.starter.files.FilesManager
+import com.softwaremill.adopttapir.starter.formatting.ProjectFormatter
+import com.softwaremill.adopttapir.template.ProjectGenerator
 
 import java.io.File
-import scala.reflect.io.Directory
 
-class StarterService(
-    config: StarterConfig,
-    projectGenerator: ProjectGenerator
-) extends FLogging {
-  def generateZipFile(starterDetails: StarterDetails): IO[File] = {
-    logger.info(s"received request: $starterDetails") *>
-      IO(projectGenerator.generate(starterDetails)).flatMap { filesToCreate =>
-        IO.blocking(newTemporaryDirectory(prefix = config.tempPrefix).toJava)
-          .bracket { tempDir =>
-            for {
-              _ <- logger.debug("created temp dir: " + tempDir.toString)
-              _ <- storeFiles(tempDir, filesToCreate)
-              _ <- FormatScalaFiles(tempDir)
-              zippedFile <- zipDirectory(tempDir)
-              _ <- increaseMetricCounter(starterDetails)
-            } yield zippedFile
-          }(release = tempDir => if (config.deleteTempFolder) deleteRecursively(tempDir) else IO.unit)
-      }
-  }
-
-  private def storeFiles(destinationFolder: File, filesToCreate: List[GeneratedFile]): IO[Unit] = IO.blocking {
-    filesToCreate.foreach { generatedFile =>
-      val file = BFile(destinationFolder.getPath, generatedFile.relativePath)
-      file.parent.createDirectories()
-      file.overwrite(generatedFile.content)
-    }
-  }
+class StarterService(pg: ProjectGenerator, fm: FilesManager, pf: ProjectFormatter) extends FLogging {
 
-  private def zipDirectory(directoryFile: File): IO[File] = IO.blocking {
-    val destination = BFile.newTemporaryFile(prefix = directoryFile.getName + "_", suffix = ".zip")
-    ZipArchiver().create(destination.path, directoryFile.toPath)
-    destination.toJava
+  def generateZipFile(starterDetails: StarterDetails): IO[File] = {
+    val rawGeneratedFiles = pg.generate(starterDetails)
```

Comment:
there is log statement missing
```
logger.info(s"received request: $starterDetails") *>
```
if we keep it then files generation should be wrapped with IO effect so that all happens on IO thread...

Reply:
Updated. Log statement is now present.